lib/WebService/PayPal/PaymentsAdvanced/Response/SecureToken.pm

```diff
@@ -82,6 +100,56 @@ sub _build_hosted_form_uri {
     );
 }
 
+# Make an HTTP request to the given URI.
+#
+# Return the response if the request is successful. If the request fails with a
+# 5xx error, retry. We retry up to the configured number of times. On
+# encountering a non-5xx and non-success response, return the response
+# immediately.
+#
+# Throw an error if we exhaust our retries.
+sub _make_http_request_with_retries {
+    my $self = shift;
+    my $uri  = shift;
+
+    my $res;
+
+    # +1 so we always try at least once.
+    ## no critic (ControlStructures::ProhibitCStyleForLoops)
+    for (
+        my $attempts_left = $self->retry_attempts + 1;
+        $attempts_left > 0;
+        $attempts_left--
+        ) {
```

Comment:
The rationale for the C loop was for the `last` check to be clearly about whether this is the last attempt we have left. However I agree that the count down required more thought. In fact I had an off by one error on my first pass writing it! I've updated it now.